database-commons/src/main/java/io/cdap/plugin/db/source/CustomBigDecimalSplitter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2025 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.db.source;
+
+import org.apache.hadoop.mapreduce.lib.db.BigDecimalSplitter;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+/**
+ * Custom implementation of {@link BigDecimalSplitter} to ensures safe and precise division of BigDecimal values while
+ * calculating split points for NUMERIC and DECIMAL types.
+ */
+public class CustomBigDecimalSplitter extends BigDecimalSplitter {
+
+  public static final int SCALE_BUFFER = 5;
```

Comment:
How is the SCALE_BUFFER determined(add a comment)? Would this value work for all scenarios ? 
If not what is the guiding principle to determine the correct scale buffer to have? If it needs to be determined externally ?

Reply:
Added the comment.  `5` gives a good enough buffer to maintain the accuracy during the division for all practical scenarios. This ensures correctness even for databases that support very large precision values. The same buffer was applied while fixing the OOM issue for the PostgreSQL Connector as well.